backend/danswer/listeners/slack_listener.py

```diff
@@ -0,0 +1,193 @@
+import os
+
+from danswer.configs.app_configs import QDRANT_DEFAULT_COLLECTION
+from danswer.connectors.slack.utils import make_slack_api_rate_limited
+from danswer.direct_qa.answer_question import answer_question
+from danswer.server.models import QAResponse
+from danswer.server.models import QuestionRequest
+from danswer.server.models import SearchDoc
+from danswer.utils.logging import setup_logger
+from slack_sdk import WebClient
+from slack_sdk.socket_mode import SocketModeClient
+from slack_sdk.socket_mode.request import SocketModeRequest
+from slack_sdk.socket_mode.response import SocketModeResponse
+
+logger = setup_logger()
+
+_NUM_RETRIES = 3
+_NUM_DOCS_TO_DISPLAY = 5
+
+
+def _get_socket_client() -> SocketModeClient:
+    app_token = os.environ.get("DANSWER_BOT_SLACK_APP_TOKEN")
```

Comment:
Please reference docs here in case anyone hits issues and looks in the code

Reply:
Added link to a page in our docs which will describe the process of setting this up